lib/event_bus/services/notification.ex

```diff
@@ -69,9 +73,19 @@ defmodule EventBus.Service.Notification do
     Logger.warn(msg)
   end
 
-  @spec log_error(module(), any()) :: no_return()
-  defp log_error(subscriber, error) do
-    msg = "#{subscriber}.process/1 raised an error!\n#{inspect(error)}"
+  @spec log_error(module(), any(), any()) :: no_return()
+  defp log_error(subscriber, error, stacktrace) do
+    msg =
+      "#{subscriber}.process/1 raised an error!\n#{
+        Exception.format_banner(:error, error, stacktrace)
+      }"
+
+    case @error_handler do
+      nil -> nil
+      {module, function} -> apply(module, function, [error, stacktrace])
+      _ -> Logger.warn("Error handler must be { Module, :function_atom }")
+    end
+
     Logger.info(msg)
```

Comment:
I think the only thing you need to change is the following line to support custom error handler:

`Logger.info(msg)` should be replaced as `Logger.error(Exception.format(:error, msg, __STACKTRACE__))`

The rest of them can/should be done via logger config, for example if you need to set the logger to Airbrake in your config:

```
config :logger,
  backends: [{Airbrake.LoggerBackend, :error}, :console]
```